### PR TITLE
Feat on3477 implement ml scorer function

### PR DESCRIPTION
**Pull request by mircorudolph**
added ML comparator as function

ml_comparator(city: dict, actionA:dict, actionB:dict) -> int

it returns -1 for action B is preferred and 1 for action A is preferred

### What change is being made?

Implement an ML-based scorer function to compare actions and integrate it into the benchmarking utility, updating dependencies to support new ML features.

### Why are these changes being made?

These changes introduce a machine learning-based method for action comparison to enhance accuracy in decision-making, leveraging an XGBoost model for predictions. The update addresses the need to improve ranking precision by utilizing advanced modeling, as described in the newly integrated `ml_comparator.py`, enhancing the functionality of the decision-making toolkit. Additionally, this necessitated updating dependencies to support the machine learning framework and functionalities.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: prioritizer/utils/ml_comparator.py [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+import xgboost as xgb
+from pathlib import Path
+
+root_path = Path(__file__).resolve().parent.parent.parent
+
+# Load the scaler from the saved file
+with open(Path(root_path / "data" / "ml" / "scaler" / "scaler.pkl"), "rb") as f:
+    scaler = pickle.load(f)
```
[the remark this answers is not in the document]
Comment (piotrnowakowski) [reply prioritizer/utils/ml_comparator.py:11]: This is our file so not important

**Review by piotrnowakowski — APPROVED** [review]

File: prioritizer/utils/ml_comparator.py [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+    # Final feature cleanup
+    df_transformed = prepare_final_features(df_transformed)
+
+    # Make a prediction with the model
+    prediction = predict_xgb(df_transformed)
+
+    return prediction
+
+
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (mircorudolph) [reply prioritizer/utils/ml_comparator.py:641]: This is just for testing. If we import the module this does not get executed
